firehose-source/src/main/java/demo/FirehoseApplication.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright 2015 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package demo;
+
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.tomcat.websocket.WsWebSocketContainer;
+import org.cloudfoundry.client.lib.CloudCredentials;
+import org.cloudfoundry.client.lib.oauth2.OauthClient;
+import org.cloudfoundry.client.lib.util.RestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpHeaders;
+import org.springframework.integration.websocket.ClientWebSocketContainer;
+import org.springframework.util.StringUtils;
+import org.springframework.web.socket.client.WebSocketClient;
+import org.springframework.web.socket.client.standard.StandardWebSocketClient;
+import source.FirehoseProperties;
+import source.FirehoseSource;
+
+import javax.net.ssl.SSLContext;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+
+/**
+ * @author Vinicius Carvalho
+ */
+@SpringBootApplication
+@EnableConfigurationProperties(FirehoseProperties.class)
+@ComponentScan(basePackageClasses = FirehoseSource.class)
```

Comment:
Once this class has changed package, the component scan will typically not be needed